lib/services/graph-progress.js

```diff
@@ -0,0 +1,232 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+'use strict';
+
+module.exports = graphProgressServiceFactory;
+
+graphProgressServiceFactory.$provide = 'Services.GraphProgress';
+graphProgressServiceFactory.$inject = [
+    '_',
+    'Protocol.Events',
+    'Services.Waterline',
+    'Logger',
+    'Constants',
+    'Promise',
+    'Assert',
+    'GraphProgress'
+];
+
+function graphProgressServiceFactory(
+    _,
+    eventsProtocol,
+    waterline,
+    Logger,
+    Constants,
+    Promise,
+    assert,
+    GraphProgress
+) {
+    var logger = Logger.initialize(graphProgressServiceFactory);
+
+    function GraphProgressService() {
+    }
+
+    GraphProgressService.prototype._publishGraphProgress = function(
+        graph,
+        graphState,
+        swallowError
+    ) {
+        assert.object(graph, 'graph');
+        assert.string(graph.name , 'graph.name');
+        assert.string(graphState, 'graphState');
+
+        var graphDescription = 'Graph "' + graph.name + '" ' + graphState;
+        var progress = GraphProgress.create(graph, graphDescription);
+
+        return eventsProtocol.publishProgressEvent(
+            graph.instanceId,
+            progress.getProgressEventData()
+        )
+        .catch(function(error) {
+            if(swallowError) {
```

Comment:
But it will decrease the readability of the API, `publishGraphProgress(graph, graphState, options)`.
When I read this definition of the function, I have to read the comments or the code to find out what is inside options.